### ISSUE TITLE
Slaves corpses can be generated under Gozag

### ISSUE THREAD

**Issue by adrien-martin** [crawl/crawl#988]
**VERSION:**
0.23

**DESCRIPTION:**
Corpses can be generated when killing Pikel slaves and worshipping Gozag.
Killed Kirke too, but don't remember anything special about her slaves.

I didn't reproduced it at the moment, is there any way to generate Pikel with his band in wizmod ?

**Comment by josbraden** [comment]
I was able to reproduce in wizard mode, but I think this is intended behavior? For Gozag to turn things to gold they have to give XP:

mon-death.cc

```
bool goldify = have_passive(passive_t::goldify_corpses)
               && mons_gives_xp(mons, you)
               && !force;
```

**Comment by alexjurkiewicz** [comment]
You can generate Pikel and band in wizmode with `&m pikel band` btw. This
works for any monster that has a potential band configured

**Comment by adrien-martin** [comment]
> You can generate Pikel and band in wizmode with `&m pikel band` btw. This works for any monster that has a potential band configured

Ah ok, can reproduce it now too, thanks!

So as a side note Kirke pigs give gold, but freed pigs can give human corpses.

> I was able to reproduce in wizard mode, but I think this is intended behavior? For Gozag to turn things to gold they have to give XP:
> 
> mon-death.cc
> 
> ```
> bool goldify = have_passive(passive_t::goldify_corpses)
>                && mons_gives_xp(mons, you)
>                && !force;
> ```

Ok, I didn't thought about the experience.
It make sense no gold is generated.
But corpses under Gozag feels surprising.

**Comment by NormalPerson7** [comment]
In reality these monsters probably shouldn't give corpses at all. In fact, there's an open bug report on Mantis that is similar. (That one concerns the inaccuracy of the "This monster grants no experience, nutrition or items statement; they grant no experience, but they do grant corpses and drop their animal skins. See https://crawl.develz.org/mantis/view.php?id=11459.)

**Comment by MarvinPA** [comment]
Fixed in 312109b4ccb1f.
